File: module-server/src/main/java/de/twomartens/wahlrecht/model/internal/ElectedResult.java
package de.twomartens.wahlrecht.model.internal;

import java.util.Collection;
import java.util.LinkedList;
import java.util.Map;
import java.util.Map.Entry;
import java.util.stream.Collectors;
import lombok.Builder;

@Builder
public record ElectedResult(Map<VotingResult, Collection<ElectedCandidate>> electedCandidates,
                            LinkedList<Double> usedElectionNumbers) {

  public Map<Nomination, Collection<ElectedCandidate>> electedCandidatesByNomination() {
    return electedCandidates.entrySet().stream()
        .collect(Collectors.toMap(entry -> entry.getKey().getNomination(), Entry::getValue));
  }

}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/model/internal/ElectedResult.kt
package de.twomartens.wahlrecht.model.internal

data class ElectedResult(
    val electedCandidates: Map<VotingResult, Collection<ElectedCandidate>>,
    val usedElectionNumbers: List<Double>
) {
    val electedCandidatesByNomination = electedCandidates.entries.asSequence()
        .map { Pair(it.key.nominationId, it.value) }
        .toMap()
}
